binfmt/libelf/libelf_load.c

```diff
@@ -96,7 +96,11 @@ static void elf_elfsize(FAR struct elf_loadinfo_s *loadinfo)
            * able
            */
 
-          if ((shdr->sh_flags & SHF_WRITE) != 0)
+          if ((shdr->sh_flags & SHF_WRITE) != 0
+#ifdef CONFIG_ARCH_HAVE_TEXT_HEAP_WORD_ALIGNED_READ
```

Comment:
should we remove this and always enable SHF_EXECINSTR check?

Reply:
Eventually yes, but I tried to be as "restrictive" as possible as I could only test (I meant HW testing) when `CONFIG_ARCH_HAVE_TEXT_HEAP_WORD_ALIGNED_READ` is enabled (which is my use case). I prefer to keep it more constraint. What do you think?

Reply:
Ok